src/SprykerEco/Service/Computop/ComputopService.php

```diff
@@ -0,0 +1,119 @@
+<?php
+
+/**
+ * MIT License
+ * Use of this software requires acceptance of the Evaluation License Agreement. See LICENSE file.
+ */
+
+namespace SprykerEco\Service\Computop;
+
+use Generated\Shared\Transfer\ComputopCreditCardPaymentTransfer;
+use Spryker\Service\Kernel\AbstractService;
+use SprykerEco\Shared\Computop\ComputopConstants;
+
+/**
+ * @method \SprykerEco\Service\Computop\ComputopServiceFactory getFactory()
+ */
+class ComputopService extends AbstractService implements ComputopServiceInterface
+{
+
+    /**
+     * @inheritdoc
+     */
+    public function getDescriptionValue(array $items)
+    {
+        return $this->getFactory()->createComputopMapper()->getDescriptionValue($items);
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function getComputopMacHashHmacValue(ComputopCreditCardPaymentTransfer $cardPaymentTransfer)
+    {
+        $value = $this->getFactory()->createComputopMapper()->getMacEncryptedValue($cardPaymentTransfer);
+        return $this->getHashHmacValue($value);
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function extractHeader(array $decryptedArray, $method)
+    {
+        $header = $this->getFactory()->createComputopConverter()->extractHeader($decryptedArray, $method);
+        $neededMac = $this->getHashHmacValue($this->getFactory()->createComputopMapper()->getMacResponseEncryptedValue($header));
```

Comment:
was does Mac mean? Do people know that? Or should this be `neededHashValue` etc?